homework-solutions/ts/hw-15/task-1.ts

```diff
@@ -0,0 +1,49 @@
+/*
+1. Создайте интерфейс IEmployee {name: string, salary: number, isManager: boolean } и объект QA: IEmployee
+*/
+
+interface IEmployee {
+    name: string, 
+    salary: number, 
+    isManager: boolean
+}
+
+const QA:IEmployee = {
+    name: "Andry",
+    salary: 100000,
+    isManager: false
+}
+
+/*
+2. Создайте тип EmployeeKeys, который будет юнионом названий ключей IEmployee (использовать keyof)
+*/
+type EmployeeKeys = keyof IEmployee
+/*
+3. Создайте тип QaKeys, который будет содержать лишь ключи объекта QA(использовать keyof и typeof)
+*/
+type QaKeys = keyof typeof QA
+/*
+4. Создайте тип UserType из объекта QA (используйте typeof)
+*/
+type UserType = typeof QA
+/*
+5. Создайте тип, в котором все поля интерфейса IEmployee будут необязательными
+*/
+type OptionIEmployee = Partial<IEmployee>
+/*
+6. Создайте тип, который будет содержать поля name и salary из интерфейса IEmployee
+*/
+type SomeOfIEmployee1 = Pick<IEmployee, 'name' | 'salary'>
+/*
+7. Создайте тип, который будет держать все поля из интерфейса IEmployee, кроме isManager
+*/
+type SomeOfIEmployee2 = Omit<IEmployee, 'isManager'>
```

Comment:
https://google.github.io/styleguide/tsguide.html#descriptive-names
```suggestion
type EmployeeBasicInfo = Pick<IEmployee, 'name' | 'salary'>
/*
7. Создайте тип, который будет держать все поля из интерфейса IEmployee, кроме isManager
*/
type EmployeeWithoutManagerFlag = Omit<IEmployee, 'isManager'>
```